### PR TITLE
tests/osd: fix typo in unittest_osdmap

### PR DESCRIPTION
**Pull request by hjwsm1989**
Signed-off-by: huangjun <huangjun@xsky.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/test/osd/TestOSDMap.cc [hunk 1]
```diff
@@ -979,7 +979,7 @@ TEST_F(OSDMapTest, CleanPGUpmaps) {
         int new_up_primary;
         osdmap.pg_to_raw_up(pgid, &new_up, &new_up_primary);
         ASSERT_TRUE(new_up == up);
-        ASSERT_TRUE(new_up_primary = up_primary);
+        ASSERT_TRUE(new_up_primary == up_primary);
```
Comment (tchaikov) [inline src/test/osd/TestOSDMap.cc]: i'd suggest use 
```c++
ASSERT_EQ(up_primary, new_up_primary);
```

File: src/test/osd/TestOSDMap.cc [hunk 2]
```diff
@@ -979,7 +979,7 @@ TEST_F(OSDMapTest, CleanPGUpmaps) {
         int new_up_primary;
         osdmap.pg_to_raw_up(pgid, &new_up, &new_up_primary);
         ASSERT_TRUE(new_up == up);
-        ASSERT_TRUE(new_up_primary = up_primary);
+        ASSERT_EQ(new_up_primary, up_primary);
```
Comment (tchaikov) [inline src/test/osd/TestOSDMap.cc]: again, i'd suggest use 
```c++
ASSERT_EQ(up_primary, new_up_primary);
```

see the definition of `ASSERT_EQ`
```c++
#define GTEST_ASSERT_EQ(expected, actual) \
  ASSERT_PRED_FORMAT2(::testing::internal:: \
                      EqHelper<GTEST_IS_NULL_LITERAL_(expected)>::Compare, \
                      expected, actual)
```
the first parameter is the expected value. in this case, it's `up_primary`.

also, could you take this opportunity to update the line above?

Comment (hjwsm1989) [reply src/test/osd/TestOSDMap.cc]: sorry, i misundestood 'the line above'

**Comment by tchaikov** [comment]
retest this please.

**Pull request merged**